### PR TITLE
Add 'join tables' for visits

### PR DESCRIPTION
**Pull request by tobuya**
- Create a table named `vets` with the following column: id INT (set it as autoincremented PRIMARY KEY), name VARCHAR, age INT, date_of_graduation DATE.
- Create a "join table" called `specializations` to handle the many-to-many relationship between the tables `species` and `vets`.
- Create a "join table" called `visits` to handle the many-to-many relationship between the tables `animals` and `vets`.
- Insert data into `vets`, `specializations`, and `visits`.
- Write a query to find the last animal seen by William Tatcher
Screenshot 2023-03-03 175312

- Write a query to find the number of different animals Stephanie Mendez see
Screenshot 2023-03-03 180334

- Write a query to list all vets and their specialties, including vets with no specialties
Screenshot 2023-03-03 182116

- Write a query to list all animals that visited Stephanie Mendez between April 1st and August 30th, 2020
Screenshot 2023-03-03 183049

- Write a query to find an animal that has the most visits to vets
Screenshot 2023-03-03 183748

- Write a query to find who was Maisy Smith's first visit
Screenshot 2023-03-03 201416

- Write a query to find details for the most recent visit: animal information, vet information, and date of visit
Screenshot 2023-03-03 202041

- Write a query to find how many visits were with a vet that did not specialize in that animal's species
Screenshot 2023-03-03 202847

- Write a query to find a specialty that Maisy Smith consider getting into based on the species she gets the most
Screenshot 2023-03-03 203714

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -57,14 +65,108 @@ JOIN owners o
...
+-- Number of visits with a vet who did not specialize in that animal's species
+SELECT COUNT(*) 
+FROM vets ve 
+JOIN visits vi 
+ON vi.vet_id = ve.id
+JOIN animals a
+ON vi.animal_id = a.id
+JOIN specializations sp 
+ON ve.id = sp.vet_id
+WHERE NOT sp.species_id = a.species_id;
```
Comment (Gambit142) [inline queries.sql]: - [ ] This query returns the wrong result. The query should return the total number of visits by animals to vets who were not specialized in the animal's species. For this, the result should be 12. This is the breakdown:

- William Thatcher who specializes in Pokemon had 2 visits by Digimon
- Jack Harkness who specializes in Digimon had 1 visit by a Pokemon
- Maisy Smith doesn't specialize in any species but had 9 visits. 

This makes a total of _**12**_ visits. Kindly write a query that selects the `visits` table that the `LEFT JOIN` the `animals` table that `LEFT JOIN` the `vets` table on `animals.species_id` that isn't in the `species_id` in the specialization table. 👍 😎

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @tobuya 👋, 

Good job so far!

## Highlights
- Everything that creates or changes the structure of the tables is in the _**schema.sql**_ file
- Everything that populates tables with data is in the _**data.sql**_ file
- Everything that is a query that helps to analyze data is in the _**queries.sql**_ file

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:

N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please, remember to tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: queries.sql (hunk 1 above)
Comment (tobuya) [reply queries.sql]: @Gambit142 Thank you for the feedback. Am on it 👍

**Review by sumairq — APPROVED** [review]
### Hi @tobuya  , 



Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Good commit history
✔️Linters are passing
✔️Very professional README
✔️Correct Git flow
✔️Descriptive PR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**